### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by rmrmrmr**
image

## On this pull request we:

- Installed the `CanCanCan` gem
- Added a `role` column to the `Users` table via an ActiveRecord Migration
- Implemented the ability for users to delete a post. 
    - We used CanCanCan for this feature to only be available to users with `role = 'admin'`
- Implemented the ability for users to delete a comment.
    - We used CanCanCan for this feature to only be available to users with `role = 'admin'`
- Refactored the `update_posts_counter` and `update_comments_counter` methods to work with the newly implemented features.

- Here are two Loom videos showcasing the new features!
    - On this [video](https://www.loom.com/share/65e5433b3cdc4ed9bda567603cd5ee70), I log in with a User with admin privileges. I can see the `Delete comment` and `Delete` buttons and I use them to delete a comment and a post 
    - On this [video](https://www.loom.com/share/41e4632da070436792dd2670f9588dfc), I log in as a User with no admin privileges. This user isn't shown any `Delete` buttons

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(user)
+    user ||= User.new
+
+    if user.admin?
+      can :destroy, Post
+      can :destroy, Comment
+    else
+      can :read, :all
+    end
```
Comment (fobadara) [inline app/models/ability.rb]: [REQUIRED] Kindly adjust your code. Users are supposed to be able to delete posts and comments if it belongs to them.

**Review by fobadara — APPROVED** [review]
Hi team, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Good first attempt
✅ Clean code

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/ability.rb (hunk 1 above)
Comment (rmrmrmr) [reply app/models/ability.rb]: Thanks, @fobadara . Almost forgot to include that condition!

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @rmrmrmr and @Juank628 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Descriptive PR.
- Good commit messages.
- Correct git-flow.

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**